core/builtin.py

```diff
@@ -604,28 +604,51 @@ def Cd(argv, mem, dir_stack):
   return 0
 
 
-WITH_PREFIX = 1
-WITHOUT_PREFIX = 2
+WITH_LINE_NUMBERS = 1
+WITHOUT_LINE_NUMBERS = 2
 SINGLE_LINE = 3
+TILDE_PREFIX_FORM = 1
+LONG_FORM = 2
 
-def _PrintDirStack(dir_stack, style):
+def _PrintDirStack(dir_stack, mem, style, form):
```

Comment:
I think the interface to this function can be simplified.  Why not just pass an optional `home_dir=None` arg?  If it's not None, then do the translation.

We don't need `mem` or `form`.  That can be pushed to the outer levels.